jupyter_book_to_htmlbook/code_processing.py

```diff
@@ -111,3 +112,67 @@ def number_codeblock(pre_block, cell_number):
                 element.replace_with(element.replace('\n', f'\n{indent}'))
 
     return cell_number
+
+
+def process_code_examples(chapter):
+    """
+    Applies appropriate data types and adds titles for formal code
+    "Examples" in the text
+    """
+    examples = chapter.find_all("div", class_="tag_example")
+
+    for example_cell in examples:
+        pre_block = example_cell.find("pre")
+        example_div = pre_block.parent
+        try:
+            comments = pre_block.find_all("span", class_="c1")
+
+            # ensure comments are within the first three spans (since we
+            # expect an empty span to start based on their highlighter)
+            if not (
+                    comments[0] in pre_block.find_all("span")[0:3] and
+                    comments[1] in pre_block.find_all("span")[0:3]
+                   ):
+                logging.warning(
+                    "Missing first two line comments for uuid and title." +
+                    f"Unable to apply example formatting to {example_cell}.")
+                return example_cell
```

Comment:
After adding this guard "if", I realize that the function became quite long, if not necessarily overly complex (maybe it is, I don't know, I can't see straight anymore). If this function is too long to be readable cleanly please holler at me.